### PR TITLE
Change the code search to sort results by relevance

### LINKED ISSUE #32129: Sort the code search result by relevance
### Feature Description

I noticed that the code search uses the `repo_id` to sort the results. I propose we start sorting by relevance (i.e., by `_score`). 

This will be needed by #32096 (I've noticed this while coding it), as GitHub's search considers matches in filenames more relevant than those in content. Moreover, Sorting by relevance seems more sensible than sorting by the `repo_id`.

If you guys are ok with it, I'll open a PR.

P.S. It was hard to decide if this was a bug or a feature. In the end, I went with the latter since the code search does work.

P.S. Both [ElasticSearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/sort-search-results.html) and [Bleve](https://blevesearch.com/docs/Sorting/) allow sorting by relevance.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by bsofiato**
Resolves #32129

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: modules/indexer/code/bleve/bleve.go [hunk 1]
```diff
@@ -284,6 +284,8 @@ func (b *Indexer) Search(ctx context.Context, opts *internal.SearchOptions) (int
 		searchRequest.AddFacet("languages", bleve.NewFacetRequest("Language", 10))
 	}
 
+	searchRequest.SortBy([]string{"-_score", "RepoID"})
```
Comment (lunny) [inline modules/indexer/code/bleve/bleve.go]: Maybe the second sort factor should be updated time but not repo_id?

**Comment by lunny** [comment]
And I found it's wired when using fuzzy search with bleve. Some fuzzy results' scores bigger than exact match results' . So is there an sort option so that exact match will be always before fuzzy results?

File: modules/indexer/code/bleve/bleve.go (hunk 1 above)
Comment (bsofiato) [reply modules/indexer/code/bleve/bleve.go]: Cool. I think it makes total sense.

I'm gonna change it use the update time instead of the repo id.

**Comment by bsofiato** [comment]
> And I found it's wired when using fuzzy search with bleve. Some fuzzy results' scores bigger than exact match results' . So is there an sort option so that exact match will be always before fuzzy results?

I can take a look into it. I'm not familiar with bleve, but I'm gonna check !! Off the top of my head, we can do an `OR` query and assign a greater weight to statement that uses exact match. 

BTW, I'm working on #32096 and I had to rework the query. Maybe I can enhance it further to boost the score of exact matches there. What do you think ?

File: modules/indexer/code/bleve/bleve.go (hunk 1 above)
Comment (bsofiato) [reply modules/indexer/code/bleve/bleve.go]: @lunny Done

**Comment by lafriks** [comment]
I think that can be improved in other PR when implementing advanced search queries

**Review by lafriks — APPROVED** [review]

**Pull request merged**